test/MarginAccount.test.ts

```diff
@@ -0,0 +1,486 @@
+  let marginAccountTester: MarginAccountTesterInstance
+  let otoken: OtokenInstance
+  let otoken2: OtokenInstance
+  let usdc: MockERC20Instance
+  let mockAddressBookAddr: string
+
+  // let expiry: number;
+  const strikePrice = new BigNumber(200).times(new BigNumber(10).exponentiatedBy(18))
+  const expiry = 1601020800 // 2020/09/25 0800 UTC
+  const isPut = true
+
+  const testAccount: {owner: string; vaultIds: BigNumber} = {owner: ZERO_ADDR, vaultIds: new BigNumber(0)}
+  let account: {owner: string; vaultIds: BigNumber}
+
+  before('Deployment', async () => {
+    // deploy WETH token
+    weth = await MockERC20.new('WETH', 'WETH')
+    // deploy AddressBook token
+    marginAccountTester = await MarginAccountTester.new()
+
+    const addressBook = await MockAddressBook.new()
+    mockAddressBookAddr = addressBook.address
+    await addressBook.setController(controller)
+
+    // deploy oToken with addressbook
+    otoken = await Otoken.new(addressBook.address)
+
+    usdc = await MockERC20.new('USDC', 'USDC')
+
+    // deploy second otoken
+    otoken2 = await Otoken.new(addressBook.address)
+    await otoken2.init(ETH_ADDR, usdc.address, usdc.address, strikePrice, expiry, isPut, {from: deployer})
+  })
+
+  describe('Otoken Initialization', () => {
+    it('should be able to initialize with put parameter correctly', async () => {
+      await otoken.init(ETH_ADDR, usdc.address, usdc.address, strikePrice, expiry, isPut, {from: deployer})
+      assert.equal(await otoken.underlyingAsset(), ETH_ADDR)
+      assert.equal(await otoken.strikeAsset(), usdc.address)
+      assert.equal(await otoken.collateralAsset(), usdc.address)
+      assert.equal((await otoken.strikePrice()).toString(), strikePrice.toString())
+      assert.equal(await otoken.isPut(), isPut)
+      assert.equal((await otoken.expiry()).toNumber(), expiry)
+    })
+
+    it('should initialize the put option with valid name / symbol', async () => {
+      assert.equal(await otoken.name(), `ETHUSDC 25-September-2020 200Put USDC Collateral`)
+      assert.equal(await otoken.symbol(), `oETHUSDC-25SEP20-200P`)
+      assert.equal((await otoken.decimals()).toNumber(), 18)
+    })
+
+    describe('Open new vault', () => {
+      it('vaultIds should be zero, owner should be null', async () => {
+        account = await marginAccountTester.getAccount({from: deployer})
+        assert.equal(account.owner, testAccount.owner, 'MarginAccount.Account owner addr mismatch')
+        assert.equal(account.vaultIds, testAccount.vaultIds, 'Incorrect vaultIds')
+      })
+
+      //TODO: should vaultIds be at 1 or 0 at the end (i.e. how do we do indices)
+      it('should increment vaultID otoken implementation address', async () => {
+        await marginAccountTester.testOpenNewVault({from: deployer})
+        const account = await marginAccountTester.getAccount({from: deployer})
+        assert.equal(account.vaultIds.toString(), '1', 'vaultIds is not 1 after new vault opened')
```

Comment:
hmm shouldn't vaultId be 0 when there is 1 vault?

Reply:
I agree we should have a test on vaults[0] before this test case.

Reply:
added tests - lmk if they are what you were thinking of

Reply:
vaultIds represents number of vaults right? In that case it should be 1